pkg/cloud/openstack/cluster/actuator.go

```diff
@@ -219,18 +221,13 @@ func (a *Actuator) storeCluster(cluster *clusterv1.Cluster, clusterCopy *cluster
 		cluster.ResourceVersion = result.ResourceVersion
 	}
 
-	// set to APIServerLoadBalancer IP & Port for now. With generated kubeadm
-	// this can be changed to clusterConfiguration.controlPlaneEndpoint for the
-	// non-LoadBalancer case. For now it doesn't matter because the Status is not
-	// used yet.
-	apiServerHost := clusterProviderSpec.APIServerLoadBalancerFloatingIP
-	apiServerPort := clusterProviderSpec.APIServerLoadBalancerPort
-	if clusterProviderSpec.ManagedAPIServerLoadBalancer {
-		if clusterProviderStatus.Network != nil && clusterProviderStatus.Network.APIServerLoadBalancer != nil &&
-			apiServerHost != clusterProviderStatus.Network.APIServerLoadBalancer.IP {
-			return fmt.Errorf("APIServerLoadBalancer has IP %s instead of IP %s", clusterProviderStatus.Network.APIServerLoadBalancer.IP, apiServerHost)
-		}
+	apiServerHost := strings.Split(clusterProviderSpec.ClusterConfiguration.ControlPlaneEndpoint, ":")[0]
```

Comment:
Small nit: could create a var above this line like `controlPlaneURI := strings.Split(clusterProviderSpec.ClusterConfiguration.ControlPlaneEndpoint, ":")` and then for apiServerHost and apiServerPortStr do `controlPlaneURI[0]` & `controlPlaneURI[1]`

Reply:
Good point. Done